### PR TITLE
span: remove finished attribute

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The `finished` attribute is not required on a span since it can be inferred based off of the existence of the `duration_ns` attribute.

To save some bytes, we can remove it.

### CODE REVIEW

File: ddtrace/span.py [hunk 1]
```diff
@@ -116,6 +112,17 @@ def start(self):
...
+    def finished(self):
+        return self.duration_ns is not None
+
+    @finished.setter
+    def finished(self, value):
+        if value is False:
+            self.duration_ns = None
+        elif value is True and not self.finished:
+            self.duration_ns = time_ns() - self.start_ns
+
```
Comment (Kyle-Verhoog) [inline ddtrace/span.py:125]: I'm not 100% about this implementation for the setter. Being able to mutate this attribute publicly doesn't really make sense in the first place. The only place we do it is to make testing some internals easier. We could instead deprecate setting the value publicly. Thoughts?

Comment (jd) [reply ddtrace/span.py:125]: I think it's a good idea and it's a nice way of finishing a trace without having to bother with the time function.

However, you currently ignore non boolean value and that's not good. Either:
- raise `ValueError` as last resort
- use `if value/else` and don't check type

Comment (Kyle-Verhoog) [reply ddtrace/span.py:125]: Good points. I did consider it but didn't think `span.finished = 342432` made much sense. The ValueError makes sense to me though!

Comment (Kyle-Verhoog) [reply ddtrace/span.py:128]: On second thought, I think the `if value/else` is a bit more pythonic. 🤷‍♂️

File: ddtrace/span.py [hunk 2]
```diff
@@ -116,6 +112,20 @@ def start(self):
...
+    def finished(self):
+        return self.duration_ns is not None
+
+    @finished.setter
+    def finished(self, value):
+        if value:
+            if not self.finished:
+                self.duration_ns = time_ns() - self.start_ns
+            else:
+                log.warning("Ignoring finish on already finished span.")
```
Comment (jd) [inline ddtrace/span.py:125]: Please don't log and hide a programming error. It makes it extremely hard to debug or even write tests.

Comment (Kyle-Verhoog) [reply ddtrace/span.py:125]: Hmm true. I think we could do away with it altogether and document that any subsequent `finish = True` will be ignored (the same logic as for .finish()`)

Comment (jd) [reply ddtrace/span.py]: Works for me. As long as you document it, I think it's 100% fine.

**Pull request merged**